src/jit/codegencommon.cpp

```diff
@@ -10486,7 +10505,23 @@ void CodeGen::genSetScopeInfo()
     }
 #endif
 
-    if (compiler->info.compVarScopesCount == 0)
+    unsigned varsLocationsCount = 0;
+
+#ifdef USING_SCOPE_INFO
+    if (compiler->info.compVarScopesCount > 0)
+    {
+        varsLocationsCount = siScopeCnt + psiScopeCnt;
+    }
+#else // USING_SCOPE_INFO
+
+#ifdef USING_VARIABLE_LIVE_RANGE
+    const VariableLiveKeeper* varLiveKeeper = compiler->getVariableLiveKeeper();
+    varsLocationsCount                      = (unsigned int)varLiveKeeper->getLiveRangesCount();
+#endif // USING_VARIABLE_LIVE_RANGE
```

Comment:
Can we change the ifdefs to assert in some way if neither of the ifdef code paths are taken. Maybe for example:

```
#ifdef USING_SCOPE_INFO

...

#elif USING_VARIABLE_LIVE_RANGE

#else

unreached();

#endif
```

Reply:
Do you mean if any of the ways of tracking variables is taken? I was assuming that as something that should happen without errors.

Reply:
Makes sense